lib/pyszn/injection.py

```diff
@@ -150,14 +154,46 @@ def parse_attribute_injection(
                     result[filename]['environment'][attribute] = value
             except KeyError:
                 pass
+
+            # Get the topology for the filename and if it doesn't exist in the cache
+            # parse it and save it to the parsed_topologies_cache
+            if filename.endswith('.py'):
+                topology = find_topology_in_python(filename, szn_dir=szn_dir)
+                if topology is None:
+                    log.warning(
+                        ('Skipping node expansion for attribute injection in filename '
+                        '{} in the lookup path as it does not contain a TOPOLOGY or TOPOLOGY_ID '
+                        'definition.').format(filename))
+                    return []
+            else:
+                with open(filename, 'r') as fd:
+                    topology = fd.read().strip()
+
+            # Check if topology exists in cache, otherwise parse it and save it
+            if topology in parsed_topologies_cache:
+                parsed_topology = parsed_topologies_cache[topology]
+            else:
+                try:
+                    parsed_topology = parse_txtmeta(topology)
+                except Exception:
+                    parsed_topology = {}
+                    log.error(
+                        ('Skipping node expansion for attribute injection in filename '
+                        '{} in the lookup path as SZN format parsing failed.'
+                        ).format(filename))
+                    log.debug(format_exc())
```

Comment:
I think if you `break` here it would be safe. Also you could delete the not so good lookin `if parsed_topology == {}:` in the _expand_ methods bellow

Reply:
I agree, that wasn't very good looking.